### PR TITLE
[POAE7-2460] Transform non-groupby-agg result to Arrow Format

### PR DESCRIPTION
**Pull request by haiweizh**
### What changes were proposed in this pull request?

Support Non Groupby Agg for Arrow Format.

### Why are the changes needed?

Non Groupby Agg result is not arrow format. Convert non-groupby-agg result to Arrow Format based on schema when fetch result.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Uts.

### Which label does this PR belong to?

cider,test

### CODE REVIEW

File: cider/tests/functionality/CiderAggTest.cpp [hunk 1]
```diff
@@ -106,6 +106,53 @@ TEST_F(CiderAggTest, sumTest) {
   // TODO: SUM(decimal) with half null
 }
 
+TEST_F(CiderAggTest, aggArrowTest) {
+  // SUM(int)
+  prepareArrowBatch();
```
Comment (yma11) [inline cider/tests/functionality/CiderAggTest.cpp:111]: `QueryArrowDataGenerator ` should can be used for data generation now. We need to switch to it, you can refer [PR](https://github.com/intel/BDTK/pull/97) first.

File: cider/exec/module/CiderRuntimeModule.cpp [hunk 2]
```diff
@@ -608,6 +608,9 @@ CiderRuntimeModule::fetchResults(int32_t max_row) {
     auto out_batch = setSchemaAndUpdateAggResIfNeed(std::move(non_groupby_agg_result));
     // reset need to be done after data consumed, like bitmap for count(distinct)
     resetAggVal();
+    if (ciderCompilationOption_.use_cider_data_format) {
```
Comment (jikunshang) [inline cider/exec/module/CiderRuntimeModule.cpp:611]: I don't think we should do this. `convertToArrowRepresentation` method should be removed and just for test usage.

File: cider/tests/functionality/CiderAggTest.cpp (hunk 1 above)
Comment (haiweizh) [reply cider/tests/functionality/CiderAggTest.cpp]: Done.

File: cider/exec/module/CiderRuntimeModule.cpp (hunk 2 above)
Comment (haiweizh) [reply cider/exec/module/CiderRuntimeModule.cpp:611]: Done. Remove convert function code to CiderBatchUtils.cpp.

**Comment by haiweizh** [comment]
PR Ready, Plz help to take a review. @yma11 @jikunshang

File: cider/tests/CiderNongroupbyAggTest.cpp [hunk 3]
```diff
@@ -188,6 +188,7 @@ void testScalarTypeSingleKey(MockTable* table,
 }
 
 TEST_F(CiderNongroupbyAggArrowFormatTest, SingleKeyScalarTypeTest) {
+  GTEST_SKIP_("FIXME(haiwei): [POAE7-2511] Update verify test case of Non Groupby Agg");
```
Comment (yma11) [inline cider/tests/CiderNongroupbyAggTest.cpp:191]: Shouldn't these TCs be resolved in this PR?

Comment (haiweizh) [reply cider/tests/CiderNongroupbyAggTest.cpp:191]: This case has a large amount of changes, so i split a subtask to update it.

**Review by yma11 — APPROVED** [review]

File: cider/exec/module/batch/CiderBatchUtils.cpp [hunk 4]
```diff
@@ -141,6 +179,7 @@ int64_t getBufferNum(const ArrowSchema* schema) {
     case 'l':
     case 'f':
     case 'g':
+    case 'd':
```
Comment (jikunshang) [inline cider/exec/module/batch/CiderBatchUtils.cpp:182]: d here means decimal type?

File: cider/exec/module/batch/CiderBatchUtils.cpp [hunk 5]
```diff
@@ -22,14 +22,52 @@
...
+
+#define GENERATE_AND_ADD_COLUMN(C_TYPE)                                       \
+  {                                                                           \
+    std::vector<C_TYPE> col_data;                                             \
+    std::vector<bool> null_data;                                              \
+    col_data.reserve(table_row_num);                                          \
+    null_data.reserve(table_row_num);                                         \
+    null_data.push_back(false);                                               \
+    C_TYPE* buf = (C_TYPE*)table_ptr[i];                                      \
+    C_TYPE* null_buf = (C_TYPE*)table_ptr[i * 2];                             \
```
Comment (jikunshang) [inline cider/exec/module/batch/CiderBatchUtils.cpp:60]: null buffer use C_TYPE to represent as well?

**Review by jikunshang — APPROVED** [review]

**Pull request merged**